src/registration/forceRegistration.ts

```diff
@@ -122,22 +124,35 @@ async function notifyState() {
   }
 }
 
+let awaiting = false;
 function awaitLoginNotification() {
+  if (awaiting) {
```

Comment:
remove all the awaiting, and just return a promise which is resolved after the notifications is shown, promise is resolved only once, but you need to keep the same instance of it

Reply:
why is that easiser / better?

Reply:
it is a more pragmatic way  of managing an async flow, as we did here :
https://github.com/codota/tabnine-vscode/blob/168deb7698863341024bd49322cc2ff915f8341c/src/capabilities/capabilities.ts#L62

Reply:
what is the syntax for that?

Reply:
this is just a completion handler. Where is it being resolved between 2 states

Reply:
@dimacodota